wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php

```diff
@@ -549,6 +553,58 @@ public static function setOfficeDetails($op, $groupID, $entityID, &$params) {
 
   }
 
+  /**
+   * This hook is called after the database write on a custom table.
+   *
+   * @param string $op
+   *   The type of operation being performed.
+   * @param string $objectName
+   *   The custom group ID.
+   * @param int $objectId
+   *   The entityID of the row in the custom table.
+   * @param object $objectRef
+   *   The parameters that were sent into the calling function.
+   */
+  public static function displayActivityDataForCollectionCamp($op, $groupID, $entityID, &$params) {
+    if ($op !== 'create') {
```

Comment:
We should place an additional check so that this function runs ONLY when a Collection Camp is created. Otherwise it will run for every entity in Civi.

```suggestion
    if ($op !== 'create' || $entityId !== 'Eck_Collection_Camp') {
```

Reply:
The function is not running because we're receiving an integer for the entity ID. This check prevents it from executing.